src/core/core.controller.js

```diff
@@ -645,10 +645,15 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 
 		// Draw all the scales
 		helpers.each(me.boxes, function(box) {
-			box.draw(me.chartArea);
+			box.draw(me.chartArea, false);
 		}, me);
 
 		me.drawDatasets(easingValue);
+
+		helpers.each(me.boxes, function(box) {
```

Comment:
I'm really not fan of this extra loop calling `draw()` a second time for all scales and as you said, it would draw existing custom scales 2 times. I don't think scales should be aware of the drawing order and check internally this extra argument "to draw or not to draw". IMO, `draw()` should be called only one time before or after datasets.